ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/MyAbstractSplitFilter.java

```diff
@@ -0,0 +1,159 @@
+package org.apache.hadoop.hive.ql.optimizer.calcite.rules;
```

Comment:
Many of these rules could be moved to Calcite instead of implementing it here. Also, do any of these rules already exist in Calcite? If some changes are needed, we should do them there. Then we could extend them here if some behavior is specific to Hive. For those that stay here, can we move them to package ```org.apache.hadoop.hive.ql.optimizer.calcite.rules.jdbc``` and rename them to ```JDBCXXXRule```?